src/routes/admin/mentor/mentor.route.test.ts

```diff
@@ -75,4 +75,18 @@ describe('Admin mentor routes', () => {
       .send({ status: 'approved' })
       .expect(403)
   })
+
+  it('should return mentors and a success message when mentors are found', async () => {
+    const response = await adminAgent
+      .get(`/api/admin/mentors?status=${ApplicationStatus.APPROVED}`)
+      .expect(200)
+
+    expect(response.body).toHaveProperty('mentors')
+  })
```

Comment:
Write a test to get the mentors without the status as well.
```suggestion
  it('should return mentors and a success message when mentors are found', async () => {
    const response = await adminAgent
      .get(`/api/admin/mentors`)
      .expect(200)

    expect(response.body).toHaveProperty('mentors')
  })
```

Reply:
```
const status: ApplicationStatus = req.query.status as ApplicationStatus

    if (user.type !== ProfileTypes.ADMIN) {
      res.status(403).json({ message: 'Only Admins are allowed' })
    } else {
      if (!(status?.toUpperCase() in ApplicationStatus)) {
        res.status(400).json({ message: 'Please provide a valid status' })
        return
      }
      const { mentors, statusCode, message } = await getAllMentors(status)
      res.status(statusCode).json({ mentors, message })
    }
```.  can u help how can i modify this code because it requires status here ,

Reply:
Can you try this?

```
export const getAllMentorsByStatus = async (
  req: Request,
  res: Response
): Promise<void> => {
  try {
    const user = req.user as Profile
    const status: ApplicationStatus | undefined = req.query.status as ApplicationStatus | undefined

    if (user.type !== ProfileTypes.ADMIN) {
      res.status(403).json({ message: 'Only Admins are allowed' })
    } else {
      if (status && !(status.toUpperCase() in ApplicationStatus)) {
        res.status(400).json({ message: 'Please provide a valid status' })
        return
      }
      const { mentors, statusCode, message } = await getAllMentors(status)
      res.status(statusCode).json({ mentors, message })
    }
  } catch (err) {
    if (err instanceof Error) {
      console.error('Error executing query', err)
      res
        .status(500)
        .json({ error: 'Internal server error', message: err.message })
    }
  }
}

```

Reply:
thanks noted